core/src/main/java/org/kohsuke/stapler/RequestImpl.java

```diff
@@ -259,17 +259,29 @@ public String getParameter(String name) {
     }
 
     @Override
-    public Map getParameterMap() {
-        Map parameterMap = super.getParameterMap();
+    public Map<String, String[]> getParameterMap() {
```

Comment:
as this is itself implementing the servlet spec, should it not also return an immutable map?

Possibly too risky for a backport, but moving forward for a weekly?

Reply:
I think it should indeed.